FullClusterRestartIT#testWatcher failure in CI

The testWatcher test in the FullClusterRestartIT class failed in CI, https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2595/console. This did not reproduce for me.

Reproduce line:

```
./gradlew :x-pack:qa:full-cluster-restart:without-system-key:v7.0.0#upgradedClusterTestRunner \
  -Dtests.seed=B937D3C214B32CC5 \
  -Dtests.class=org.elasticsearch.xpack.restart.FullClusterRestartIT \
  -Dtests.method="testWatcher" \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-TN \
  -Dtests.timezone=America/Swift_Current \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Error:

```
java.lang.AssertionError: 
Expected: <2>
     but: was <3>
	at __randomizedtesting.SeedInfo.seed([B937D3C214B32CC5:D4DA200FE4D2BB33]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.restart.FullClusterRestartIT.assertBasicWatchInteractions(FullClusterRestartIT.java:369)
	at org.elasticsearch.xpack.restart.FullClusterRestartIT.testWatcher(FullClusterRestartIT.java:186)
```

It looks like the test depends on versioning, but something else updates the watch between creation and the update; maybe execution of the watch?

--- a/x-pack/qa/full-cluster-restart/src/test/java/org/elasticsearch/xpack/restart/FullClusterRestartIT.java
+++ b/x-pack/qa/full-cluster-restart/src/test/java/org/elasticsearch/xpack/restart/FullClusterRestartIT.java
@@ -116,7 +116,6 @@ public class FullClusterRestartIT extends AbstractFullClusterRestartTestCase {
     }
 
     @SuppressWarnings("unchecked")
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/40178")
     public void testWatcher() throws Exception {
         if (isRunningAgainstOldCluster()) {
             logger.info("Adding a watch on old cluster {}", getOldClusterVersion());

